relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -216,149 +167,146 @@ mod tests {
+    }
     #[test]
     /// Tests that an event is dropped when there is a match and we have 0% sample rate.
-    fn test_get_sampling_result_return_drop_with_match_and_0_sample_rate() {
-        let project_state = project_state_with_config(SamplingConfig {
-            rules: vec![],
-            rules_v2: vec![mocked_sampling_rule(1, RuleType::Transaction, 0.0)],
-            mode: SamplingMode::Received,
-        });
-        let event = mocked_event(EventType::Transaction, "transaction", "2.0");
-
-        let result = get_sampling_result(true, Some(&project_state), None, None, Some(&event));
-        assert_eq!(
-            result,
-            SamplingResult::Drop(MatchedRuleIds(vec![RuleId(1)]))
-        )
+    fn test_match_rules_return_drop_with_match_and_0_sample_rate() {
+        let event = mocked_event(EventType::Transaction, "bar", "2.0");
+        let rules = vec![mocked_sampling_rule(1, RuleType::Transaction, 0.0)];
+        let seed = Uuid::default();
+
+        let result: SamplingResult = SamplingEvaluator::new(Utc::now())
+            .match_rules(seed, &event, rules.iter())
+            .into();
+
+        assert!(result.is_match());
+        assert!(result.should_drop());
     }
 
     #[test]
     /// Tests that an event is kept when there is no match.
-    fn test_get_sampling_result_return_keep_with_no_match() {
-        let project_state = project_state_with_config(SamplingConfig {
-            rules: vec![],
-            rules_v2: vec![SamplingRule {
-                condition: eq("event.transaction", &["foo"], true),
-                sampling_value: SamplingValue::SampleRate { value: 0.5 },
-                ty: RuleType::Transaction,
-                id: RuleId(3),
-                time_range: Default::default(),
-                decaying_fn: Default::default(),
-            }],
-            mode: SamplingMode::Received,
-        });
+    fn test_match_rules_return_keep_with_no_match() {
+        let rules = vec![SamplingRule {
+            condition: eq("event.transaction", &["foo"], true),
+            sampling_value: SamplingValue::SampleRate { value: 0.5 },
+            ty: RuleType::Transaction,
+            id: RuleId(3),
+            time_range: Default::default(),
+            decaying_fn: Default::default(),
+        }];
+
         let event = mocked_event(EventType::Transaction, "bar", "2.0");
+        let seed = Uuid::default();
+
+        let result: SamplingResult = SamplingEvaluator::new(Utc::now())
+            .match_rules(seed, &event, rules.iter())
+            .into();
 
-        let result = get_sampling_result(true, Some(&project_state), None, None, Some(&event));
-        assert_eq!(result, SamplingResult::Keep)
+        assert!(result.is_no_match());
+        assert!(result.should_keep());
     }
 
     #[test]
-    /// Tests that an event is kept when there are unsupported rules with no processing and vice versa.
-    fn test_get_sampling_result_return_keep_with_unsupported_rule() {
```

Comment:
replaced by `fn test_is_trace_fully_sampled_return_true_with_unsupported_rules()` and `test_matching_with_unsupported_rule`